src/api/api.js

```diff
@@ -1,12 +1,25 @@
 import axios from "axios";
 
-export function getArticles(id) {
-  const URL = "https://nc-news-u31g.onrender.com/api/articles";
-  if (id) {
-    return axios.get(`${URL}/${id}`);
-  } else {
-    return axios.get(URL);
-  }
+export function getArticles(page) {
+  return axios.get(`https://nc-news-u31g.onrender.com/api/articles`, {
+    params: {
+      p: page,
+    },
+  });
+}
+
+export function getArticlesById(article_id) {
+  return axios.get(
+    `https://nc-news-u31g.onrender.com/api/articles/${article_id}`
+  );
+}
+
+export function getArticlesByTopic(topic) {
+  return axios.get(`https://nc-news-u31g.onrender.com/api/articles`, {
```

Comment:
We should be able to reuse the original getArticles function here, if topic is undefined it will simply ignore it and get all articles.